js-pkg/create-y-sweet-app/src/templates/nextjs/components/Presence.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import Image from "next/image";
+import { usePresence, usePresenceSetter } from "@y-sweet/react";
+import { useEffect, useState } from "react";
+import { randomColor } from "@/lib/colors";
+
+export function Presence() {
+  const setPresence = usePresenceSetter();
+  const presence = usePresence();
+
+  const [self, setSelf] = useState(() => ({ color: randomColor() }));
+  useEffect(() => setPresence(self), [setPresence, self]);
```

Comment:
One thing we should do in the React library is make the `setPresence` function (and others) referentially stable so it doesn't force `useEffect`s to rerun unnecessarily. (I _think_ it's not referentially stable right now.)